src/azure-cli/azure/cli/command_modules/eventgrid/advanced_filter.py

```diff
@@ -62,36 +83,89 @@ def __call__(self, parser, namespace, values, option_string=None):
                            " a valid operator. Supported values are ".format(operator) +
                            NUMBERIN + "," + NUMBERNOTIN + "," + STRINGIN + "," +
                            STRINGNOTIN + "," + STRINGBEGINSWITH + "," +
                            STRINGCONTAINS + "," + STRINGENDSWITH + "," +
                            NUMBERGREATERTHAN + "," + NUMBERGREATERTHANOREQUALS + "," +
-                           NUMBERLESSTHAN + "," + NUMBERLESSTHANOREQUALS + "," + BOOLEQUALS + ".")
+                           NUMBERLESSTHAN + "," + NUMBERLESSTHANOREQUALS + "," + BOOLEQUALS + "," +
+                           NUMBERINRANGE + "," + NUMBERNOTINRANGE + "," +
+                           ISNULLORUNDEFINED + "," + ISNOTNULL + "," +
+                           STRINGNOTBEGINSWITH + "," + STRINGNOTENDSWITH + "," +
+                           STRINGNOTCONTAINS + ".")
         if namespace.advanced_filter is None:
             namespace.advanced_filter = []
         namespace.advanced_filter.append(advanced_filter)
 
 
+def _get_in_advanced_filter(key, operator, values):
+    _validate_at_least_single_value_is_specified(operator, values)
+    float_values = [float(i) for i in values[2:]]
+
+    if operator.lower() == NUMBERIN.lower():
+        advanced_filter = NumberInAdvancedFilter(key=key, values=float_values)
+    elif operator.lower() == NUMBERNOTIN.lower():
+        advanced_filter = NumberNotInAdvancedFilter(key=key, values=float_values)
+
+    return advanced_filter
+
+
+def _get_in_range_advanced_filter(key, operator, values):
+    result = []
+    for value in values[2:]:
+        float_value = [float(i) for i in value.split(',')]
+        result.append(float_value)
+
+    if operator.lower() == NUMBERINRANGE.lower():
+        advanced_filter = NumberInRangeAdvancedFilter(key=key, values=result)
+    elif operator.lower() == NUMBERNOTINRANGE.lower():
+        advanced_filter = NumberNotInRangeAdvancedFilter(key=key, values=result)
+    return advanced_filter
+
+
 def _validate_only_single_value_is_specified(operator_type, values):
     if len(values) != 3:
         raise CLIError("--advanced-filter: For '{}' operator, only one filter value "
                        "must be specified.".format(operator_type))
+
+
+def _validate_no_value_is_specified(operator_type, values):
+    if len(values) != 2:
+        raise CLIError("--advanced-filter: For '{}' operator, no filter value "
+                       "must be specified.".format(operator_type))
+
+
+def _validate_at_least_single_value_is_specified(operator_type, values):
+    if len(values) < 3:
+        raise CLIError("--advanced-filter: For '{}' operator at least one filter value "
+                       "must be specified.".format(operator_type))
+
+
+def _validate_values_len(values):
+    valuesLen = len(values)
+    if valuesLen < 2:
+        raise CLIError('usage error: --advanced-filter KEY[.INNERKEY] FILTEROPERATOR VALUE [VALUE...]')
```

Comment:
How about modify the check that for `IsNullOrUndefined` and `IsNotNull`, `valuesLen` should not be less than `2`, and for all others `valuesLen` should not be less than `3`.

Since all others AdvancedFilter take `value=value[2:]`, to avoid add `_validate_at_least_single_value_is_specified` for each AdvancedFilter, it's better to check here. How do you think?

Reply:
per offline conversation breaking into diff methods.